lib/auth/clusterconfig/clusterconfigv1/service_test.go

```
@@ -1755,3 +1764,111 @@ func newTestEnv(opts ...serviceOpt) (*env, error) {
+			},
+		},
+		{
+			name:              "authorized proxy with non empty access graph config; Policy module is disabled",
+			role:              types.RoleProxy,
+			accessGraphConfig: cfgEnabled,
+			errorAssertion:    require.NoError,
+			responseAssertion: &clusterconfigpb.GetClusterAccessGraphConfigResponse{
+				AccessGraph: &clusterconfigpb.AccessGraphConfig{
+					Enabled: false,
+				},
+			},
+		},
+		{
+			name: "authorized proxy with non empty access graph config; Policy module is enabled",
+			role: types.RoleProxy,
+			testSetup: func(t *testing.T) {
+				m := modules.TestModules{
+					TestFeatures: modules.Features{
+						Policy: modules.PolicyFeature{
+							Enabled: true,
+						},
+					},
+				}
+				modules.SetTestModules(t, &m)
+			},
+			accessGraphConfig: cfgEnabled,
+			errorAssertion:    require.NoError,
+			responseAssertion: &clusterconfigpb.GetClusterAccessGraphConfigResponse{
+				AccessGraph: &clusterconfigpb.AccessGraphConfig{
+					Enabled:  true,
+					Insecure: true,
+					Address:  "address",
+					Ca:       []byte("ca"),
+				},
+			},
+		},
+		{
+			name: "authorized discovery with non empty access graph config; Policy module is enabled",
+			role: types.RoleDiscovery,
+			testSetup: func(t *testing.T) {
+				m := modules.TestModules{
+					TestFeatures: modules.Features{
+						Policy: modules.PolicyFeature{
+							Enabled: true,
+						},
+					},
+				}
+				modules.SetTestModules(t, &m)
+			},
+			accessGraphConfig: cfgEnabled,
+			errorAssertion:    require.NoError,
+			responseAssertion: &clusterconfigpb.GetClusterAccessGraphConfigResponse{
+				AccessGraph: &clusterconfigpb.AccessGraphConfig{
+					Enabled:  true,
+					Insecure: true,
+					Address:  "address",
+					Ca:       []byte("ca"),
+				},
+			},
+		},
+	}
+
+	for _, test := range cases {
+		t.Run(test.name, func(t *testing.T) {
+			if test.testSetup != nil {
```

nit: I'd remove this check and let the test panic if the setup function is not set. Panic here is less evil than forgetting to set the setup function, even if it's empty.